### PR TITLE
Upstream libc++ buildbot restarter.

### PR DESCRIPTION
**Pull request by EricWF**
I've been running a cronjob on my local machine to restart preempted
libc++ CI runs. This is bad and brittle. This upstreams a much better
version of the restarter.

It works by matching on check run annotations looking for mention
of the machine being shutdown.

If there are both preempted jobs and failing jobs, we don't restart
the workflow. Maybe we should change that?

### CODE REVIEW

**Comment by EricWF** [comment]
@tstellar Once this lands we can revoke the permissions for the r[libcxx-buildbot-restarter](https://github.com/apps/libcxx-buildbot-restarter) app.

File: .github/workflows/restart-preempted-libcxx-jobs.yaml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    types:
+      - failure
+      - canceled
+
+permissions:
+  contents: read
+
+jobs:
+  restart:
+    name: "Restart Job"
```
Comment (tstellar) [inline .github/workflows/restart-preempted-libcxx-jobs.yaml:25]: We need to disable this workflow for forks, so
` if: github.repository_owner == 'llvm'`

**Review by tstellar — APPROVED** [review]
LGTM.

**Comment by EricWF** [comment]
I've deleted the restarter app.

**Pull request merged**